controllers/register.go

```diff
@@ -0,0 +1,61 @@
+package controllers
+
+import (
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/go-playground/validator/v10"
+	errorHandler "github.com/gutakk/go-google-scraper/helpers/error_handler"
+	session "github.com/gutakk/go-google-scraper/helpers/session"
+	"github.com/gutakk/go-google-scraper/models"
+	"golang.org/x/crypto/bcrypt"
+	"gorm.io/gorm"
+)
+
+type RegisterController struct {
+	DB *gorm.DB
+}
+
+type RegisterForm struct {
+	Email           string `form:"email" binding:"email,required"`
+	Password        string `form:"password" binding:"required,min=6"`
+	ConfirmPassword string `form:"confirm-password" binding:"eqfield=Password,required"`
+}
+
+func (r *RegisterController) applyRoutes(engine *gin.Engine) {
+	engine.GET("/register", r.displayRegister)
+	engine.POST("/register", r.register)
+}
+
+func (r *RegisterController) displayRegister(c *gin.Context) {
+	c.HTML(http.StatusOK, "register.html", gin.H{
+		"title": "Register",
+	})
+}
+
+func (r *RegisterController) register(c *gin.Context) {
+	form := &RegisterForm{}
+
+	if err := c.ShouldBind(form); err != nil {
+		for _, fieldErr := range err.(validator.ValidationErrors) {
+			c.HTML(http.StatusBadRequest, "register.html", gin.H{
+				"title": "Register",
+				"error": errorHandler.ValidationErrorToText(fieldErr),
```

Comment:
Did you mean email should resent to template even the error is `email already exists`?

Reply:
In any case of errors on any form, it's better to fill in the form with the user-submitted data so that they can understand what they did wrong. For passwords, since it's sensitive, we should not.

Reply:
Fixed in 4e2d7e17370093d8bdab769c3f8bd302a73f73f4